synapse/storage/databases/main/stream.py

```diff
@@ -1616,3 +1616,48 @@ async def get_name_from_instance_id(self, instance_id: int) -> str:
             retcol="instance_name",
             desc="get_name_from_instance_id",
         )
+
+    async def get_timeline_gaps(
+        self,
+        room_id: str,
+        from_token: Optional[RoomStreamToken],
+        to_token: RoomStreamToken,
+    ) -> Optional[RoomStreamToken]:
+        """Check if there is a gap, and return the token."""
```

Comment:
"the token" sounds like it can be expanded a bit to say whether this is the start of the gap, etc. (see the implementation comments at the end for calculating the `last_position`)

Reply:
```suggestion
        """Check if there is a gap, and return a token that marks the position
        of the gap in the stream.
        """
```

Like that?

Reply:
That's better. Thanks!